File: common/src/main/java/se/casaferre/common/Volume.java
package se.casaferre.common;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.experimental.Accessors;

import java.io.Serializable;

/**
 * Purpose of this class is to map volumes
 * <p>
 * Created by Jorgen Andersson (jorgen@kollektiva.se) on 2018-01-29.
 */
@Accessors(prefix = "m")
@AllArgsConstructor
public class Volume implements Serializable {

    private static final Double LITER_2_US_GALLON = 3.78541;
    private static final Double LITER_2_UK_GALLON = 4.54609;

    @Getter
    private Double mLiter;
    @Getter
    private Double mUsGallon;
    @Getter
    private Double mUkGallon;

    public static Volume setLiter(Double volume) {
        return new Volume(
                volume,
                volume / LITER_2_US_GALLON,
                volume / LITER_2_UK_GALLON);
    }

    public static Volume setUsGallon(Double volume) {
        return new Volume(
                volume * LITER_2_US_GALLON,
                volume,
                volume * (LITER_2_US_GALLON / LITER_2_UK_GALLON));
    }

    public static Volume setUkGallon(Double volume) {
        return new Volume(
                volume * LITER_2_UK_GALLON,
                volume * (LITER_2_UK_GALLON / LITER_2_US_GALLON),
                volume);
    }
}

File: common/src/main/kotlin/se/casaferre/common/Volume.kt
package se.casaferre.common

/**
 * Purpose of this file is ...
 *
 * Created by Jorgen Andersson on 2018-06-29.
 */
object Volume {

    fun liter2usGallon(value: Double): Double {
        return value / 3.78541
    }

    fun liter2imperialGallon(value: Double): Double {
        return value / 4.54609
    }

    fun usGallon2liter(value: Double): Double {
        return value * 3.78541
    }

    fun imperialGallon2liter(value: Double): Double {
        return value * 4.54609
    }
}